src/validation.cpp

```diff
@@ -5143,6 +5143,7 @@ double GuessVerificationProgress(const ChainTxData& data, const CBlockIndex *pin
 }
 
 Optional<uint256> ChainstateManager::SnapshotBlockhash() const {
+    LOCK(::cs_main);  // for m_active_chainstate access
```

Comment:
I don't find these `// for m_active_chainstate access` comments very useful. This function is _only_ accessing `m_active_chainstate` so it's pretty obvious what the lock is for.

Reply:
Removed!